src/shared/containers/ResourceViewContainer.tsx

```diff
@@ -608,14 +606,64 @@ const ResourceViewContainer: React.FunctionComponent<{
     { key: 'jira', name: 'jira', pluginComponent: jiraPlugin },
     { key: 'analysis', name: 'Analysis', pluginComponent: analysisPlugin },
   ];
-  React.useEffect(() => {
+
+  useEffect(() => {
     return () => {
       dispatch({
         type: UISettingsActionTypes.UPDATE_CURRENT_RESOURCE_VIEW,
         payload: null,
       });
     };
   }, []);
+
+  function constructUnDeprecateUrl(
+    apiEndpoint: string,
+    resource: Resource,
+    latestResource: Resource,
+    orgLabel: string,
+    projectLabel: string
+  ) {
+    return `${apiEndpoint}/${
+      resource!['@type'] === 'File' ? 'files' : 'resources'
+    }/${orgLabel}/${projectLabel}/${
+      resource!['@type'] === 'File' ? '' : '_/'
+    }${encodeURIComponent(resource!['@id'])}/undeprecate?rev=${
+      latestResource!._rev
+    }`;
+  }
+
+  const { mutate: unDeprecateResource } = useMutation({
+    mutationFn: async () => {
+      try {
+        await nexus.httpPut({
+          path: constructUnDeprecateUrl(
+            apiEndpoint,
+            resource!,
+            latestResource!,
+            orgLabel,
+            projectLabel
+          ),
+        });
+
+        setLatestResource({
+          ...latestResource!,
+          _rev: latestResource!._rev + 1,
+          _deprecated: false,
+        });
+
+        goToResource(orgLabel, projectLabel, resourceId, {
+          revision: latestResource!._rev + 1, // Go to the n + 1 = latest revision after the un-deprecation
+        });
```

Comment:
no. need to specify the revision, when not specifying the revision the new page will fetch the latest revision by default

Reply:
I quickly re-tested it, and I found out that this is not true. 
<img width="1186" alt="image" src="https://github.com/BlueBrain/nexus-web/assets/34251194/b621d723-6501-4bb4-93c7-a30a4827a295">

When not performing this action, the component's state will be updated, but React does not automatically rerender the component. To ensure that it is rendered, I enforce the update. Since it is a state based on the route, it is preferable to increment it by 1 and truly force the update. I could adjust the helper function or just do a +1, my thoughts were that the +1 is cleaner.

Reply:
may be you get me wrong, when redirecting you don't have to set the revision 
meaning:
`goToResource(orgLabel, projectLabel, resourceId, {})`

so in brief, no need to state the state because you are redirecting and no need to specify the revision because the page will fetch the latest by default

Reply:
Ah, indeed, I misunderstood you there! That does make sense. I have fixed it, thanks!